tensorboard/proto_test.py

```diff
@@ -0,0 +1,104 @@
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import difflib
+import importlib
+
+import tensorflow as tf
+from google.protobuf import descriptor_pb2
+
+
+PROTO_IMPORTS = [
+  ('tensorflow.core.framework.allocation_description_pb2', 'tensorboard.proto.allocation_description_pb2'),
+  ('tensorflow.core.framework.api_def_pb2', 'tensorboard.proto.api_def_pb2'),
+  ('tensorflow.core.framework.attr_value_pb2', 'tensorboard.proto.attr_value_pb2'),
+  ('tensorflow.core.protobuf.cluster_pb2', 'tensorboard.proto.cluster_pb2'),
+  # TODO: Enable this. Seems like we have a mismatch with use_run_handler_pool added
+  # ('tensorflow.core.protobuf.config_pb2', 'tensorboard.proto.config_pb2'),
+  ('tensorflow.core.framework.cost_graph_pb2', 'tensorboard.proto.cost_graph_pb2'),
+  ('tensorflow.python.framework.cpp_shape_inference_pb2', 'tensorboard.proto.cpp_shape_inference_pb2'),
+  ('tensorflow.core.protobuf.debug_pb2', 'tensorboard.proto.debug_pb2'),
+  ('tensorflow.core.util.event_pb2', 'tensorboard.proto.event_pb2'),
+  ('tensorflow.core.framework.function_pb2', 'tensorboard.proto.function_pb2'),
+  ('tensorflow.core.framework.graph_pb2', 'tensorboard.proto.graph_pb2'),
+  ('tensorflow.core.protobuf.meta_graph_pb2', 'tensorboard.proto.meta_graph_pb2'),
+  ('tensorflow.core.framework.node_def_pb2', 'tensorboard.proto.node_def_pb2'),
+  ('tensorflow.core.framework.op_def_pb2', 'tensorboard.proto.op_def_pb2'),
+  ('tensorflow.core.framework.resource_handle_pb2', 'tensorboard.proto.resource_handle_pb2'),
+  # TODO: Enable this. Seems like we have a mismatch with pin_to_host_optimization,
+  # disable_meta_optimizer, and meta_optimizer_timeout_ms added
+  # ('tensorflow.core.protobuf.rewriter_config_pb2', 'tensorboard.proto.rewriter_config_pb2'),
+  ('tensorflow.core.protobuf.saver_pb2', 'tensorboard.proto.saver_pb2'),
+  ('tensorflow.core.framework.step_stats_pb2', 'tensorboard.proto.step_stats_pb2'),
+  ('tensorflow.core.framework.summary_pb2', 'tensorboard.proto.summary_pb2'),
+  ('tensorflow.core.framework.tensor_pb2', 'tensorboard.proto.tensor_pb2'),
+  ('tensorflow.core.framework.tensor_description_pb2', 'tensorboard.proto.tensor_description_pb2'),
+  ('tensorflow.core.framework.tensor_shape_pb2', 'tensorboard.proto.tensor_shape_pb2'),
+  ('tensorflow.core.profiler.tfprof_log_pb2', 'tensorboard.proto.tfprof_log_pb2'),
+  ('tensorflow.core.framework.types_pb2', 'tensorboard.proto.types_pb2'),
+  ('tensorflow.core.framework.versions_pb2', 'tensorboard.proto.versions_pb2'),
+]
+
+PROTO_REPLACEMENTS = [
+  ('tensorflow/core/framework/', 'tensorboard/proto/'),
+  ('tensorflow/core/protobuf/', 'tensorboard/proto/'),
+  ('tensorflow/core/profiler/', 'tensorboard/proto/'),
+  ('tensorflow/python/framework/', 'tensorboard/proto/'),
+  ('tensorflow/core/util/', 'tensorboard/proto/'),
+  ('package: "tensorflow.tfprof"', 'package: "tensorboard"'),
```

Comment:
This rename isn't present in update.sh - should it be added there?

Reply:
It's interesting, but I think this is because it depends on where the protos are converted to `_pb2.py` files. In the `update.sh` case this doesn't need to get changed. I did find another replacement that needed to be done in that file, though, so it was a good thing to look at.